### PR TITLE
Spook sect

### PR DESCRIPTION
**Pull request by Blevruz**
## About The Pull Request

Adds the God of the Dead sect, which focuses on interacting with ghosts!
Instead of having a favor system, it relies on a cooldown system and the chaplain having a number of ghost orbiting them.
It adds two new rites:
- Haunt Item, which requires one ghost orbiting the chaplain and an item to be deposited on the altar. This gives the item the ability to hold a spirit, like the spirit sword, with a 60 second cooldown.
- Dream Communion, which requires ten ghosts orbiting the chaplain and a volunteer to be buckled on the altar. This makes the volunteer fall asleep, taking suffocation damage, but also allows them to hear the voices of the dead. It has a 5 minute cooldown.

(also makes the "spirit holding" component unique since we might try to apply it multiple times)

## Why It's Good For The Game

Expands on the chaplain's role of dealing with the dead by giving him more options to communicate with them (beyond just the Camera Obscura).

## Changelog

:cl:
add: Added the God of the Dead sect - do the bidding of the bickering dead!
/:cl:

### CODE REVIEW

File: code/modules/religion/rites.dm [hunk 1]
```diff
@@ -741,3 +741,124 @@
...
+	sleeper.SetSleeping(100)
+	ADD_TRAIT(sleeper, TRAIT_SIXTHSENSE, "chaplain sect")
+	
+	sleeper.visible_message(span_notice("[sleeper] has fallen into a deep slumber..."))
+	
+	
+	while(!QDELETED(religious_tool) && !QDELETED(sleeper))
+		if(HAS_TRAIT(sleeper, TRAIT_CRITICAL_CONDITION))
+			break
+		user.apply_damage(0.1, OXY)
```
Comment (ATH1909) [inline code/modules/religion/rites.dm:823]: bruh they're just gonna outheal this with natural breathing

Comment (Blevruz) [reply code/modules/religion/rites.dm:823]: oh shit true

**Comment by ATH1909** [comment]
also, uh, did you give them a way to wake up/end the sleeping ritual without being critted?

**Comment by Blevruz** [comment]
> 
> 
> also, uh, did you give them a way to wake up/end the sleeping ritual without being critted?

no thats on the chap, you gotta trust him a lot if you wanna know whats going on in dchat

**Comment by ATH1909** [comment]
> > also, uh, did you give them a way to wake up/end the sleeping ritual without being critted?
> 
> no thats on the chap, you gotta trust him a lot if you wanna know whats going on in dchat

okay but how does the chap end the ritual

**Comment by Blevruz** [comment]
> 
> 
> > > also, uh, did you give them a way to wake up/end the sleeping ritual without being critted?
> > 
> > 
> > no thats on the chap, you gotta trust him a lot if you wanna know whats going on in dchat
> 
> okay but how does the chap end the ritual

it should check if the user is buckled but i forgot

**Comment by Coastfront** [comment]
I gotta be honest, 5 minute cooldown on the godlike power of having all the knowledge that deadchat wants to provide you seems REALLY fuckin short. 
All it would take is a single salty ghost going to the chaplain, chappie doing the ritual, and then the ghost ousting the changeling/traitor/insert stealthy antagonist here that killed them.
I gotta be honest, i think it'd be better if the sec was more based around the undead types of races, or even on resurrecting the dead, but giving a player acess to deadchat is probably one of the strongest things in the game.

**Comment by ATH1909** [comment]
> I gotta be honest, 5 minute cooldown on the godlike power of having all the knowledge that deadchat wants to provide you seems REALLY fuckin short. All it would take is a single salty ghost going to the chaplain, chappie doing the ritual, and then the ghost ousting the changeling/traitor/insert stealthy antagonist here that killed them. i think it'd be better if the sect was more based around the undead types of races, or even on resurrecting the dead, but giving a player acess to deadchat is probably one of the strongest things in the game.

a 5 minute cooldown, a willing living subject, **and 10 cooperative ghosts**, which is the rub. if you've got 10 cooperative ghosts hanging around, you probably have a murderboner on the loose anyway.

**Comment by Coastfront** [comment]
> a 5 minute cooldown, a willing living subject, **and 10 cooperative ghosts**, which is the rub. if you've got 10 cooperative ghosts hanging around, you probably have a murderboner on the loose anyway.

Alternate scenario: There is a malf ai or just other antagonists in a round, a wizard spawns or there is a murderboner, murderboner kills a handful people, before getting killed by sec, and the corpses are taken to get revived.
Before said corpses get revived, they orbit the chaplain and the chaplain decides that he wants to do funny deadchat ritual, and gets an assistant to help.
The ghosts then oust every single other antagonist currently in the round, including the malf ai or whatever. The chaplain tells sec, sec decides to do a bagsearch on everyone listed and check the ai's sat, turns out ghosts didnt lie on some and/or all of them, and now the next 20-30 minutes is uneventful because all the antagonists got wiped by sec who have dchat knowledge.
Could this sec be used to manipulate people into getting killed, both in favor of the ghosts and in favor of the guy doing the ghost reading (lying about who the ghosts said was evil etc, or the ghosts just being dicks) Yes, it could.
Will it more likely be used by chaplains who like to suck secs cock to get metaknowledge about antags for validhunting? yes. If anything this will just make antagonists start checking the chapel to see what sect they picked, and then promptly murdering the chaplain and burning down the chapel if they see spook table.

**Comment by Blevruz** [comment]
> 
> 
> > a 5 minute cooldown, a willing living subject, **and 10 cooperative ghosts**, which is the rub. if you've got 10 cooperative ghosts hanging around, you probably have a murderboner on the loose anyway.
> 
> Alternate scenario: There is a malf ai or just other antagonists in a round, a wizard spawns or there is a murderboner, murderboner kills a handful people, before getting killed by sec, and the corpses are taken to get revived. Before said corpses get revived, they orbit the chaplain and the chaplain decides that he wants to do funny deadchat ritual, and gets an assistant to help. The ghosts then oust every single other antagonist currently in the round, including the malf ai or whatever. The chaplain tells sec, sec decides to do a bagsearch on everyone listed and check the ai's sat, turns out ghosts didnt lie on some and/or all of them, and now the next 20-30 minutes is uneventful because all the antagonists got wiped by sec who have dchat knowledge. Could this sect be used to manipulate people into getting killed, both in favor of the ghosts and in favor of the guy doing the ghost reading (lying about who the ghosts said was evil etc, or the ghosts just being dicks) Yes, it could. Will it more likely be used by chaplains who like to suck secs cock to get metaknowledge about antags for validhunting? yes. If anything this will just make antagonists start checking the chapel to see what sect they picked, and then promptly murdering the chaplain and burning down the chapel if they see spook table.

I was thinking the whole trust thing on top of the relative difficulty of getting all the components together would probably suffice to make it balanced, but thats a reasonable concern. Not sure what to do with it, can try adding more drawbacks (like auditory hallucinations while hearing dchat) or reducing the time you have to hear the dead.

**Comment by Coastfront** [comment]
> > > a 5 minute cooldown, a willing living subject, **and 10 cooperative ghosts**, which is the rub. if you've got 10 cooperative ghosts hanging around, you probably have a murderboner on the loose anyway.
> > 
> > 
> > Alternate scenario: There is a malf ai or just other antagonists in a round, a wizard spawns or there is a murderboner, murderboner kills a handful people, before getting killed by sec, and the corpses are taken to get revived. Before said corpses get revived, they orbit the chaplain and the chaplain decides that he wants to do funny deadchat ritual, and gets an assistant to help. The ghosts then oust every single other antagonist currently in the round, including the malf ai or whatever. The chaplain tells sec, sec decides to do a bagsearch on everyone listed and check the ai's sat, turns out ghosts didnt lie on some and/or all of them, and now the next 20-30 minutes is uneventful because all the antagonists got wiped by sec who have dchat knowledge. Could this sect be used to manipulate people into getting killed, both in favor of the ghosts and in favor of the guy doing the ghost reading (lying about who the ghosts said was evil etc, or the ghosts just being dicks) Yes, it could. Will it more likely be used by chaplains who like to suck secs cock to get metaknowledge about antags for validhunting? yes. If anything this will just make antagonists start checking the chapel to see what sect they picked, and then promptly murdering the chaplain and burning down the chapel if they see spook table.
> 
> I was thinking the whole trust thing on top of the relative difficulty of getting all the components together would probably suffice to make it balanced, but thats a reasonable concern. Not sure what to do with it, can try adding more drawbacks (like auditory hallucinations while hearing dchat) or reducing the time you have to hear the dead.

If i may make a reccomendation, You could replace just having complete direct deadchat acess with coming up with a more indirect/less on the nose way the chaplain could communicate with the ghosts via rituals, as with anything that communicates with deadchat, it less tends to be an issue of getting ghosts to say relaible information, since its usually just guessing which is lying, its that most methods to acess deadchat are more indirect/harder to powergame. Camera obscura for example, Sure. You could spraypaint the entire fucking alphabet and take pictures of letters after asking questions, or with the ghost sword, have ghosts point to said letters.  But its very indirect, and makes it harder for the ghosts to speak, forcing alot more to be up for interpretation.

You could make it so that the ghosts, any ghosts, can only relay one word messages to the sleeper at a time during it, Its not easy to regulate what 10 ghosts are saying at once, but its doable. But if 10 ghosts, Are all trying to seperately tell you something, one word at a time, with no way to tell who's word is who's, That'd make it alot harder for the ghosts to give SUPER obvious information, While also leaving stuff up for the sleeper and the chaplain to interpret what da spirits TM want them to do.

This is an idea that can work, as long as whatever method you use to make the deadchat seyonce less direct/harder to super powergame.

**Comment by Blevruz** [comment]
> 
> 
> > > > a 5 minute cooldown, a willing living subject, **and 10 cooperative ghosts**, which is the rub. if you've got 10 cooperative ghosts hanging around, you probably have a murderboner on the loose anyway.
> > > 
> > > 
> > > Alternate scenario: There is a malf ai or just other antagonists in a round, a wizard spawns or there is a murderboner, murderboner kills a handful people, before getting killed by sec, and the corpses are taken to get revived. Before said corpses get revived, they orbit the chaplain and the chaplain decides that he wants to do funny deadchat ritual, and gets an assistant to help. The ghosts then oust every single other antagonist currently in the round, including the malf ai or whatever. The chaplain tells sec, sec decides to do a bagsearch on everyone listed and check the ai's sat, turns out ghosts didnt lie on some and/or all of them, and now the next 20-30 minutes is uneventful because all the antagonists got wiped by sec who have dchat knowledge. Could this sect be used to manipulate people into getting killed, both in favor of the ghosts and in favor of the guy doing the ghost reading (lying about who the ghosts said was evil etc, or the ghosts just being dicks) Yes, it could. Will it more likely be used by chaplains who like to suck secs cock to get metaknowledge about antags for validhunting? yes. If anything this will just make antagonists start checking the chapel to see what sect they picked, and then promptly murdering the chaplain and burning down the chapel if they see spook table.
> > 
> > 
> > I was thinking the whole trust thing on top of the relative difficulty of getting all the components together would probably suffice to make it balanced, but thats a reasonable concern. Not sure what to do with it, can try adding more drawbacks (like auditory hallucinations while hearing dchat) or reducing the time you have to hear the dead.
> 
> If i may make a reccomendation, You could replace just having complete direct deadchat acess with coming up with a more indirect/less on the nose way the chaplain could communicate with the ghosts via rituals, as with anything that communicates with deadchat, it less tends to be an issue of getting ghosts to say relaible information, since its usually just guessing which is lying, its that most methods to acess deadchat are more indirect/harder to powergame. Camera obscura for example, Sure. You could spraypaint the entire fucking alphabet and take pictures of letters after asking questions, or with the ghost sword, have ghosts point to said letters. But its very indirect, and makes it harder for the ghosts to speak, forcing alot more to be up for interpretation.
> 
> You could make it so that the ghosts, any ghosts, can only relay one word messages to the sleeper at a time during it, Its not easy to regulate what 10 ghosts are saying at once, but its doable. But if 10 ghosts, Are all trying to seperately tell you something, one word at a time, with no way to tell who's word is who's, That'd make it alot harder for the ghosts to give SUPER obvious information, While also leaving stuff up for the sleeper and the chaplain to interpret what da spirits TM want them to do.
> 
> This is an idea that can work, as long as whatever method you use to make the deadchat seyonce less direct/harder to super powergame.

interesting ideas, i think i might be able to implement something along the line of a vote for a word or a collaborative spelling thing. i'll have to give it a shot

**Comment by JohnFulpWillard** [comment]
Rite to turn someone deadchat controllable or riot

**Comment by Coastfront** [comment]
> Rite to turn someone deadchat controllable or riot

I agree so that we can kidnap the captain and make him bend to the voices of **the people**

File: code/modules/religion/rites.dm [hunk 2]
```diff
@@ -741,3 +741,124 @@
...
+	ADD_TRAIT(sleeper, TRAIT_SIXTHSENSE, "chaplain sect")
+	
+	sleeper.visible_message(span_notice("[sleeper] has fallen into a deep slumber..."))
+	
+	
+	while(!QDELETED(religious_tool) && !QDELETED(sleeper) && (sleeper in movable_reltool.buckled_mobs))
+		if(HAS_TRAIT(sleeper, TRAIT_CRITICAL_CONDITION))
+			break
+		user.apply_damage(0.5, BRUTE)
+		sleeper.SetSleeping(100)
```
Comment (tralezab) [inline code/modules/religion/rites.dm:824]:
```suggestion
		sleeper.SetSleeping(10 SECONDS)
```

File: code/modules/religion/rites.dm [hunk 3]
```diff
@@ -741,3 +741,124 @@
...
+
+/datum/religion_rites/spook/haunt_item/proc/apply_spirit_holding(obj/item/possessed_item)
+	possessed_item.AddComponent(/datum/component/spirit_holding)
+	possessed_item.name = "eerie [possessed_item.name]"				//todo: avoid eerie eerie items
+
+/datum/religion_rites/spook/haunt_item
+	name = "Haunt Item"
+	desc = "Lets the dead possess an offering."
+	ritual_length = 5 SECONDS
+	ritual_invocations =list("To those in wait...")
```
Comment (tralezab) [inline code/modules/religion/rites.dm:841]:
```suggestion
	ritual_invocations = list("To those in wait...")
```

File: code/modules/religion/rites.dm [hunk 4]
```diff
@@ -741,3 +741,124 @@
...
+	invoke_msg = "We offer a new shell!"
+	ghosts_needed = 1
+	rite_cooldown_amt = 60 SECONDS
+///the item we want to spookify
```
Comment (tralezab) [inline code/modules/religion/rites.dm:845]: indent this to be at the same level as the var it's documenting

**Review by tralezab — CHANGES_REQUESTED** [review]
here's a review

**Comment by tralezab** [comment]
as for the design, I'm not a huge fan. It doesn't really change how you play in any meaningful way. I like the idea of a death related god and I think that could make for good roleplay but the sect itself has no progression or sacrifice requirements that put people in the role as "the grim death god guy"

I see cooldowns for rites as cool and fine but not in the sense that it replaces favor entirely, also since these cost ghosts orbiting you as a requirement it feels like a lot of these rounds with chaplain aren't going to feel great if nobody follows you.

File: code/datums/components/spirit_holding.dm [hunk 5]
```diff
@@ -8,6 +8,8 @@
 	var/attempting_awakening = FALSE
 	///mob contained in the item.
 	var/mob/living/simple_animal/shade/bound_spirit
+	///a thing can only get this component once
+	dupe_mode = COMPONENT_DUPE_UNIQUE
```
Comment (Fikou) [inline code/datums/components/spirit_holding.dm:12]: vars that already exist on an object should be before new vars

**Comment by panvxv** [comment]
What about the forgotten material- hauntium? Being able to finally control this material or giving dead players ways to interact with it would interesting. As for now it is not use much and giving it bigger purpose than just being annoying would be great.
As I remember to make it you need give picture of a ghost to the Sheet-meister 2000, if nothing was changed.

**Comment by Blevruz** [comment]
> 
> 
> as for the design, I'm not a huge fan. It doesn't really change how you play in any meaningful way. I like the idea of a death related god and I think that could make for good roleplay but the sect itself has no progression or sacrifice requirements that put people in the role as "the grim death god guy"
> 
> I see cooldowns for rites as cool and fine but not in the sense that it replaces favor entirely, also since these cost ghosts orbiting you as a requirement it feels like a lot of these rounds with chaplain aren't going to feel great if nobody follows you.

the core of the idea was that you'd have to rely on getting ghosts to play nice with you, so i think that very last point is alright.
an option for a point based approach would be that you'd gain points as long as you have ghosts orbiting you, which would mean keeping an audience even when you're not using a rite.
another option would be to sacrifice hauntium or some such but thats not very interesting.

im open to suggestions but my ability to actually implement things is limited.

also 

> 
> 
> What about the forgotten material- hauntium? Being able to finally control this material or giving dead players ways to interact with it would interesting. As for now it is not use much and giving it bigger purpose than just being annoying would be great. As I remember to make it you need give picture of a ghost to the Sheet-meister 2000, if nothing was changed.

that could be neat, not sure how you'd control it though. maybe give hauntium items offered for haunting get the democracy move thing, but that'd take away their ability to attack.

**Comment by Sheits** [comment]
> Rite to turn someone deadchat controllable or riot

please...

File: code/modules/religion/rites.dm [hunk 6]
```diff
@@ -741,3 +741,124 @@
...
+	var/mob/living/carbon/human/sleeper
+	for(var/i in movable_reltool.buckled_mobs)
+		if(istype(i,/mob/living/carbon/human))
+			sleeper = i
+			break
+	if (!sleeper)
+		return FALSE
+	
+	sleeper.SetSleeping(100)
+	ADD_TRAIT(sleeper, TRAIT_SIXTHSENSE, "chaplain sect")
```
[the remark this answers is not in the document]
Comment (Blevruz) [reply code/modules/religion/rites.dm:815]: it worked on my test server at least, so maybe people were just being quiet

**Comment by ATH1909** [comment]
btw, if the rituals for this sect don't already create an alert that causes ghosts to orbit you, I'd recommend adding a ritual for that (or making it a default feature of the other rituals)

**Comment by Blevruz** [comment]
> 
> 
> btw, if the rituals for this sect don't already create an alert that causes ghosts to orbit you, I'd recommend adding a ritual for that (or making it a default feature of the other rituals)

yeah alerts are probably a good idea, will try if the maintainers are ok with the whole orbits + cooldown system

File: code/datums/components/spirit_holding.dm (hunk 5 above)
Comment (tralezab) [reply code/datums/components/spirit_holding.dm]: ty ty

File: code/modules/religion/rites.dm (hunk 6 above)
[the remark this answers is not in the document]
Comment (JohnFulpWillard) [reply code/modules/religion/rites.dm]: People near death can hear the dead (if config allows), so unless they do something differently I don't see why this wouldnt work.

**Comment by optimumtact** [comment]
I don't think allowing ghost communication is a good idea, no matter how long it takes.

Our round meta just isn't setup that way, and too many of our antags rely on stealth to survive.

**Comment by Blevruz** [comment]
> 
> 
> I don't think allowing ghost communication is a good idea, no matter how long it takes.
> 
> Our round meta just isn't setup that way, and too many of our antags rely on stealth to survive.

if the issue is that itd be too easy to convey powerful information, what would you think of something like Coastfront's idea in the large sense, by restricting what infos they can give and making it harder for them to convey them without being organised?

for instance what would you think of having the deadchat vote on words picked from the traitor codeword list or something like that?

**Comment by Nebulacrity** [comment]
> I don't think allowing ghost communication is a good idea, no matter how long it takes.
> 
> Our round meta just isn't setup that way, and too many of our antags rely on stealth to survive.

What if said communication is done in such a way that the medium hears only a single word* out of a sentence said by a ghost, and that ghost gets an internal cooldown that prevents their sentences being chosen for this communication for, say, 5-10 seconds? Would be fun to try and construct a coherent sentence out of the random words you've heard in your trance.

*The word (no numbers or other characters allowed, only letters) should have a maximum letter count, perhaps 8-10. If the selected word contains more than that amount of letters, only a random part of it is sent to the medium, with that part's letter count within the allowed amount, and with the sliced out parts being appended by hyphens.

- Example of random word selection: 

                  Ghost says: "The Chaplain is a ling, you're so fucked."
                  Medium hears: "...ling..."

- Example of large word dissection:

                  Ghost says: "DeanIvanovhasadesword"
                  Medium hears: "...-ovhasades-..."

**Comment by KittyNoodle** [comment]
Yes, this all the way, make it like a dchat controlled dream

**Comment by Wallemations** [comment]
> for instance what would you think of having the deadchat vote on words picked from the traitor codeword list or something like that?

As long as it's not _every single word_ from the traitor codeword list at once I think this is the better option. maybe 3 or 4 words chosen at once at random. 
If you still want to do the dream, maybe use the existing dream flufftext as a baseline, but instead of being random it's like a Mad Lib for the ghosts to vote on using a wordbank randomly chosen from the previously mentioned traitor codeword list? That seems like a good option to me personally, it makes it so they can't have too much power while still being able to work together for _something_.

**Comment by tralezab** [comment]
remove ghost communication!

**Review by Twaticus — CHANGES_REQUESTED** [review]
convertaltar-yellow got hit with a stray pixel :(

**Comment by Blevruz** [comment]
> 
> 
> remove ghost communication!

what i'll be doing is replacing the whole "hear dchat while sleeping" with the following

1. while the player is asleep, sequences of characters of a certain length are picked up from the deadchat at a regular interval. this limits their ability to convey useful information but can still allow namedropping.
2. while the player is asleep, deadchat gets to vote on words picked from some of the codewords files at a regular interval. this stops namedropping but may make the rite useless depending on the word selection.
3. while the player is asleep, words are picked up from the deadchat if they are in one of the codewords files. this stops namedropping and allows a wider selection of words, but risks being useless with an unorganised deadchat.

which have been OK'd by oranges
ill try to implement the 3 and have it be governed with a config thing if i manage

**Comment by CocaColaTastesGood** [comment]
isnt there already a 2tc item that lets you get ghost info pretty easily anyways?

**Comment by ExcessiveUseOfCobblestone** [comment]
The problem with interacting with the dead is that they are given a plethora of knowledge that can absolutely btfo the round (which is why things that access the dead are generally rare and usually on the antag side since they already know the "mode", at least when secret existed)

Conceptually I cannot sub to the sect to any degree if the design is going to be interacting with player-ghosts.

**Comment by Blevruz** [comment]
putting on draft until the stuff is done

**Comment by ExcessiveUseOfCobblestone** [comment]
just RE-PR when ready :)

**Pull request closed without merging**